pkg/kubectl/resource/builder_test.go

```diff
@@ -843,7 +846,7 @@ func TestIgnoreStreamErrors(t *testing.T) {
 	singular := false
 
 	err := b.Do().IntoSingular(&singular).Visit(test.Handle)
-	if err != nil || singular || len(test.Infos) != 2 {
+	if err == nil || singular || len(test.Infos) != 2 {
```

Comment:
@feihujiang I guess this is related to the core of this PR. It's quite hard for me to review especially I cannot play with the code of this PR because it needs a rebase. I will take another look tomorrow.